nixos/modules/installer/tools/nixos-install.sh

```diff
@@ -132,7 +138,9 @@ else
     system_root=$closure
     # Create a temporary file ending in .closure (so nixos-prepare-root knows to --import it) to transport the store closure
     # to the filesytem we're preparing. Also delete it on exit!
-    nix-store --export $(nix-store -qR $closure) > $system_closure
+    # Run in background to avoid blocking while trying to write to the FIFO
+    # $system_closure refers to
+    nix-store --export $(nix-store -qR $closure) > $system_closure &
```

Comment:
`nixos-prepare-root` is already using some 1.12 features for copying, but it's probably worth revising the overall mechanism to better take advantage of 1.12. It should also help with the space issues.